snappass/main.py

```diff
@@ -177,9 +171,17 @@ def handle_password():
 
 
 @app.route('/<password_key>', methods=['GET'])
-def show_password(password_key):
-    if not request_is_valid(request):
+def preview_password(password_key):
+    password_key = url_unquote_plus(password_key)
+    password = password_exists(password_key)
+    if not password:
```

Comment:
`password` isn't the password in this case. It's a boolean indicating whether or not the password exists.

I'd suggest just writing:

```python
if not password_exists(password_key):
```

Reply:
Nice catch, fixed